doc/concepts/data-concepts/index.rst

```diff
@@ -0,0 +1,38 @@
+.. _data-concepts:
+
+Data Concepts
+=============
+
+Pachyderm data concepts describe version-control primitives that
+you interact with when you use Pachyderm.
+
+Fundamentally, these concepts are very similar to the Git version-control
+system with a few notable exceptions. Because Pachyderm
+deals not only with plain text but also with binary files and with
+large datasets, it does not process the data in the same way as Git.
+When you use Git, you store a copy of the repository on your
+local machine. You work with that copy, apply your changes, and
+then send the changes to the upstream master copy of the repository
+where it gets merged.
+
+The Pachyderm version control works slightly differently. In Pachyderm,
+only a centralized repository exists and you do not store any local copies
+of that repository. Therefore, the merge, in the traditional Git meaning,
+does not occur.
+
+Instead, your data can be continuously updated in the master branch of
+your repo, while you can experiment with specific data commits in a
+separate branch or branches. Because of this behavior, you cannot
+run into a merge conflict with Pachyderm.
+
```

Comment:
We should add a paragraph here that describes the hierarchy of the data nouns in Pachyderm. Pachyderm's data versioning system has 5 main concepts:

"Repository (aka: repo) -- This is highest level data object in Pachyderm. Typically, each data set in Pachyderm is its own repo. 

Commit -- A commit is an immutable snapshot of a repo at a point in time. 

Branch -- A branch is alias (aka: tag) for a specific commit that can be automatically moved

File -- Files (and directories) are the actual data in your repo. Pachyderm works with any pretty much any type, size, or number of files. 

Provenance -- Provenance is one of the most powerful and unique concepts in Pachyderm. It expresses the relationship between/across various commits, branches, and repos. In typical data workflows, data sets are transformed to create other data sets. In Pachyderm, provenance stores that dependency graph so you can know, for example, that `commit5` in repo `bar` was derived from `commit17` in repo foo. "

Notes:
1. At the very least, the above concepts and TOC tree should definitely be the order above. 
2. Users hate reading too much. I like the idea of having the 1-sentence description of a concept in this opening section as a TLDR and then they can go to the others to learn more.
3. My provenance writeup above is possibly too long and certainly the wording can be improved I think. We should find a balance between giving them the "provenance 101" so they get the overall idea, but not overdo it.